javascript/code-challenges/challenges-09.test.js

```diff
@@ -0,0 +1,322 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that finds the maximum value in an array
+using the 'reduce' method.
+
+E.g. [4,2,7,5,9,2] -> 9
+------------------------------------------------------------------------------------------------ */
+const maxInArray = (arr) => {
+  return arr.reduce((acc, curr) => {
+    if(curr > acc) {
+      acc = curr;
+    }
+    return acc;
+  });
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named getCourseKeys that takes in the courseInfo object and returns an array containing the keys for the courseInfo object.
+
+For example: (['name', 'duration', 'topics', 'finalExam']).
+------------------------------------------------------------------------------------------------ */
+const courseInfo = { name: 'Code 301', duration: { dayTrack: '4 weeks', eveningTrack: '8 weeks'},
+  topics: ['SMACSS', 'APIs', 'NodeJS', 'SQL', 'jQuery', 'functional programming'],
+  finalExam: true
+};
+
+const getCourseKeys = (obj) => {
+  let result = [];
+  for(let key in obj) {
+    result.push(key);
+  }
+  return result;
+};
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named checkValues that takes in an object and a value and returns true if the value is in the object.
+
+
+------------------------------------------------------------------------------------------------ */
+
+const checkValues = (obj, value) => {
+  for(let key in obj) {
+    let val = obj[key];
+    if(val === value) {
+      return true;
+    }
+  }
```

Comment:
Much in the same way that there are various array methods, you also have object methods. Look into some of them to figure out how you might be able to solve this quicker. Hint: Object.values & Object.includes